### DISCUSSION TITLE
Charge kW decimal (#.##)

### DISCUSSION THREAD

**Discussion by prussell69** [teslamate-org/teslamate#2966] [Q&A]
is it possible to get the charging kW's in watts, or at least display 1 or 2 decimal place values? All I see from the query is whole numbers (integers) of kilowatts. I would like to see e.g. 2.6 kW, instead 2 kW, or 3 kW.

**Comment by DrMichael** [comment]
What dashboard are you talking of?

**Comment by prussell69** [comment]
On the teslamate "overview" dashboard
![image](https://user-images.githubusercontent.com/28122849/199829931-a2f54a0b-efb2-4448-90fe-eccc8cae834d.png)

**Comment by DrMichael** [comment]
Ah, probably yes. Hard to test, when you're not charging. But are 100 Watt mor or less really make a difference?

**Comment by prussell69** [comment]
Well, yes. It does to me. My charging follows my excess solar, so it helps to see whether it's 5.1kW, or 5.9kW.
When I ran the query in the explorer, it only returned whole numbers.

**Comment by DrMichael** [comment]
Try this...
[overview.zip](https://github.com/adriankumpf/teslamate/files/9943577/overview.zip)

**Comment by prussell69** [comment]
Thanks. Got the same result as I did. Even though it's set to show decimals, it still shows whole valves, just with a zero after the decimal place. It looks like the data coming from the original tesla data transfer is reading it as whole numbers. I even looked at the amperage data, and it's the same way. It looks like the Tesla database read is where the change would need to happen. I will investigate it further.

**Comment by DrMichael** [comment]
Ah, you're right. Stored in the database as smallint, no decimals.
